radis/lbl/calc.py

```diff
@@ -563,6 +568,10 @@ def _check_molecules_are_consistent(
             diluent=diluent,
             **kwargs_molecule,
         )
+
+        if is_Multi:
+                diluent[molecule] = fraction
```

Comment:
instead of modifying the main diluent dictionary; and re-correcting it afterwards within your loop ; why don't you just create a new add-hoc diluent dictionary specifically for this molecule ?  Seems more straightforward. And modifying input variables is always risky. 

i.E. 
``` 
diluent_for_this_molecule = {k:v for k,v in diluent.items()}  # copy 
diluent_for_this_molecule